pkgs/development/interpreters/python/cpython/2.7/default.nix

```diff
@@ -55,6 +55,13 @@ let
 
       ./properly-detect-curses.patch
 
+      # Patch from http://bugs.python.org/issue1222585 adapted to work with
+      # `patch -p1' and with last hunk removed
+      # Upstream distutils is calling C compiler to compile C++ code, which
+      # only works for GCC and Apple Clang. This makes distutils to call C++
+      # compiler when needed.
+      ./python-2.7-distutils-C++.patch
+
```

Comment:
> You can see in the original issue http://bugs.python.org/issue1222585 that distutils were considered frozen at the time, so this has became a won't fix. This Arfrever person was maintaining it outside of the source tree. The patch is supposed to fix things that get compiled with non-gcc on any platform, but the main target is of course Darwin.

If its not accepted upstream, then I think we should only use it when it is necessary. If I understand correctly, it is necessary on Darwin, so I suggest using an `optional stdenv.isDarwin` for appending this patch and the others as well.